### ISSUE TITLE
bug: @ControllerAdvice with @GraphQlExceptionHandler no longer working.

### ISSUE THREAD

**Issue by ayushs-2k4** [Netflix/dgs-framework#2017]
## Expected behavior
Should work as previously.

## Actual behavior
Now it is not calling @GraphQlExceptionHandler annotated Exception handler in @ControllerAdvice class.

## Steps to reproduce
Just update DGS dependency to 9.1.2

It was working until 9.1.1

**Comment by mrvaruntandon** [comment]
Hi @ayushs-2k4, could you explain this issue a little more?

**Comment by ayushs-2k4** [comment]
Sure, like on 9.1.1, if I query 

> query KontestQuery {
    kontestQuery {
        kontests(page: 0, perPage: 10) {


then response is coming out as 

> {
    "errors": [
        {
            "message": "Invalid page number: Invalid page number. Page number must be greater than 0.",
            "extensions": {
                "details": "Invalid page number. Page number must be greater than 0.",
                "code": "KONTESTS_INVALID_PAGE_NUMBER",
                "errorType": "UNAVAILABLE",
                "errorDetail": "SERVICE_ERROR",
                "serviceName": "KontestAPI",
                "exception": {
                    "message": "Invalid page number: Invalid page number. Page number must be greater than 0.",
                    "locations": []
                }
            }
        }
    ],
    "data": {
        "kontestQuery": {
            "kontests": null
        }
    }
}



but on 9.1.2 for same query, response is coming out as (I just changed the dependency version from 9.1.1 to 9.1.2, and nothing else):

> {
    "errors": [
        {
            "message": "com.ayushsinghal.KontestAPI.exception.InvalidPageNumberException: Invalid page number. Page number must be greater than 0.",
            "path": [
                "kontestQuery",
                "kontests"
            ],
            "extensions": {
                "errorType": "INTERNAL",
                "serviceName": "KontestAPI",
                "code": "DOWNSTREAM_SERVICE_ERROR",
                "exception": {
                    "message": "com.ayushsinghal.KontestAPI.exception.InvalidPageNumberException: Invalid page number. Page number must be greater than 0.",
                    "locations": [
                        {
                            "line": 1,
                            "column": 48
                        }
                    ],
                    "path": [
                        "kontestQuery",
                        "kontests"
                    ]
                }
            }
        }
    ],
    "data": {
        "kontestQuery": {
            "kontests": null
        }
    }
}



I don't think this is intended behavior in 9.1.2 (I cant see any in change-log)

**Comment by ls-urs-keller** [comment]
Looking at the diff:
https://github.com/Netflix/dgs-framework/compare/v9.1.1...v9.1.2#diff-4770158ecf0bfc29eb19ac7669ba624c69b6775253212f30dda3a5a0ef5dc499R124

In v9.1.1 graphql.execution.SimpleDataFetcherExceptionHandler would be configured through the default int the builder. 
In v9.1.2 com.netflix.graphql.dgs.exceptions.DefaultDataFetcherExceptionHandler is configured through the injection

You can override the bean like below to get the behaviour from v9.1.1. Not sure the change was wanted.
```
    @Bean
    DataFetcherExceptionHandler dataFetcherExceptionHandler() {
        return new SimpleDataFetcherExceptionHandler();
    }
```

**Comment by paulbakker** [comment]
Thanks for reporting this and thanks for the hint @ls-urs-keller. The previous fix exposed an existing issue.
Fixed in #2040.
